SignalServiceKit/src/Messages/Interactions/TSMessage.m

```diff
@@ -277,6 +281,35 @@ - (BOOL)shouldUseReceiptDateForSorting
     return YES;
 }
 
+#pragma mark - Update Methods
+
+// This method does the work for the "updateWith..." methods.  Please see
+// the header for a discussion of those methods.
+- (void)applyChangeToSelfAndLatestMessage:(YapDatabaseReadWriteTransaction *)transaction
```

Comment:
TIOLI 

I got a little confused between this method and the outgoing message version `applyChangeToSelfAndLatestOutgoingMessage` since the names are so similar. Especially if we continue to tweak this method, we'll have to remember to change that logic in all places. And I imagine we'll apply this pattern in more and more places as time goes on.

WDYT about implementing this in TSYapDatabaseModel, and having the changeBlock take a generic (id) param. We'd lose the compiler type-checking on the block params, but I don't like copying around these none-trivial blocks of code and comments.

A middle ground could be that the implementation lives on TSYapDatabase, and any class where we'd like to take advantage of type checking can wrap the method with a type-specific flavor.

Reply:
Done.  I also applied `NS_ASSUME_NONNULL` to `TSYapDatabaseObject`.  I suggest you take a look at that - although I don't think its controversial.

Reply:
Nice, thanks for doing both those things.